File: 卡码/代码随想录/C++/贪心/03-困难/LeetCode_0763.cpp
//763. 划分字母区间
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	vector<int> partitionLabels(string s) {
		int hash[30]={0};
		vector<int> res;
		int left=0,right=0;
		for(int i=0;i<s.length();i++){
			hash[s[i]-'a']=i;
		}
		for(int i=0;i<s.length();i++){
			right=max(right,hash[s[i]-'a']);
			if(i==right){
				res.push_back(right-left+1);
				left=i+1;
			}
		}
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/Python/贪心/03-困难/LeetCode_0763.py
#763. 划分字母区间
from typing import *

class Solution:
  def partitionLabels(self, s: str) -> List[int]:
    left,right=0,0
    hash:List[int]=[0]*30
    res:List[int]=[]
    for i in range(len(s)):
      hash[ord(s[i])-ord('a')]=i
    for i in range(len(s)):
      right=max(right,hash[ord(s[i])-ord('a')])
      if i==right:
        res.append(right-left+1)
        left=i+1
    return res